src/v/cluster_link/model/types.h

```diff
@@ -90,13 +127,59 @@ struct metadata
     using paused_t = ss::bool_class<struct paused_tag>;
     /// Flag indicating if the panda link has been paused
     paused_t paused{paused_t::no};
+    /// The state of the link
+    link_state state;
 
     friend bool operator==(const metadata&, const metadata&) = default;
 
-    auto serde_fields() { return std::tie(name, uuid, connection, paused); }
+    auto serde_fields() {
+        return std::tie(name, uuid, connection, paused, state);
+    }
+};
+
+struct add_mirror_topic_cmd
+  : serde::envelope<
+      add_mirror_topic_cmd,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    /// Name of the topic
+    ::model::topic topic;
+    /// Initial state of the topic
+    mirror_topic_state state{mirror_topic_state::active};
+
+    friend bool
+    operator==(const add_mirror_topic_cmd&, const add_mirror_topic_cmd&)
+      = default;
+
+    auto serde_fields() { return std::tie(topic, state); }
+};
+
+struct update_mirror_topic_state_cmd
```

Comment:
for these new commands, care to add a high level description of them Since this PR is mostly boilerplate, it is not clear how these will be used and who will use them (is this internal or a proxy for a user operation?)